lib/matplotlib/tests/test_image.py

```diff
@@ -230,14 +245,13 @@ def test_cursor_data():
 
 @image_comparison(baseline_images=['image_clip'])
 def test_image_clip():
-    from math import pi
-
-    fig = plt.figure()
-    ax = fig.add_subplot(111, projection='hammer')
+    d = [[1, 2], [3, 4]]
 
-    d = [[1,2],[3,4]]
+    fig, ax = plt.subplots()
+    im = ax.imshow(d)
+    patch = patches.Circle((0, 0), radius=1, transform=ax.transData)
+    im.set_clip_path(patch)
 
-    im = ax.imshow(d, extent=(-pi,pi,-pi/2,pi/2))
```

Comment:
The old test used the hammer projection, but it was completely wrong and accidental that it ever worked.  When the projection is correctly applied on the extents (-pi,pi,-pi/2,pi/2), it maps to the null BBox, since -pi and pi map to the same value.

This PR opens the door for properly supporting these sort of projections properly, but doesn't fully enable it -- these sort of trig wraparound issues will have to be resolved first.  I didn't want the scope of this PR to get out of hand (since it's really about fixing the color issues), so I thought it best to remove a test that was based on "accidental" working and replace it with something that tests what the test was truly trying to test, which is clipping an image to a non-rectangular clip path.